src/ripple/app/consensus/RCLConsensus.cpp

```diff
@@ -802,7 +802,7 @@ RCLConsensus::Adaptor::validate(
     auto v = std::make_shared<STValidation>(
         lastValidationTime_,
         validatorKeys_.publicKey,
-        validatorKeys_.secretKey,
+        *validatorKeys_.secretKey,
```

Comment:
I'm not sure if this is the correct way to solve this problem. I'm assuming that `ValidatorKeys::SecretKey` is set before this line is executed.

The said optional is constructed inside the `ValidatorKeys` constructor at https://github.com/XRPLF/rippled/blob/66bfe909e6de0dd1c7be0d9c01e088d3fed7f0f3/src/ripple/app/misc/impl/ValidatorKeys.cpp#L29